src/Block/Service/AbstractBlockService.php

```diff
@@ -41,11 +41,14 @@ abstract class AbstractBlockService implements BlockServiceInterface
     public function __construct($twigOrDeprecatedName = null, ?Environment $twig = null)
     {
         if (!$twigOrDeprecatedName instanceof Environment && 0 !== strpos(static::class, __NAMESPACE__.'\\')) {
+            $class = 'c' === static::class[0] && 0 === strpos(static::class, "class@anonymous\0") ? get_parent_class(static::class).'@anonymous' : static::class;
```

Comment:
This looks very strange oO

Reply:
This is already part of 3.x branch:
https://github.com/sonata-project/SonataBlockBundle/blob/38619fa25eda56d32361b6a50e2c5eb6b27b1891/src/Block/Service/AbstractBlockService.php#L61

It's required given the way PHP returns the class name for an anonymous class. See https://github.com/symfony/symfony/pull/29389#discussion_r237944028 for some context.